src/main/groovy/org/pih/warehouse/outbound/ImportPackingListItem.groovy

```diff
@@ -0,0 +1,98 @@
+        // without this, we would then search e.g. for quantity available to promise for a product in the default bin location
+        if (!location && source['binLocation'] && !source['binLocation'].equalsIgnoreCase(Constants.DEFAULT_BIN_LOCATION_NAME)) {
+            // We want to indicate, that a bin location for given name has not been found.
+            obj.binLocationFound = false
+        }
+        return location
+     })
+    Location binLocation
+
+    Integer quantityPicked
+
+    @BindUsing({ obj, source -> Person.findPersonByNameOrEmail(source['recipient'])})
+    Person recipient
+
+    /**
+     * Flag to indicate whether binLocation has been found and bound.
+     * We can't rely on binLocation = null, because we accept the null as DEFAULT,
+     * so we want to know if the binLocation is null, because we have not provided (or provided DEFAULT) the bin location name
+     * or the binLocation for such name does not exist.
+     * This can't be done in the BindUsing and rejectValue, because the errors would be cleared before running the validator logic
+     * The flag is set to false if a bin location is not found for given name in the BindUsing of binLocation
+     */
+    boolean binLocationFound = true
+
+
+    static constraints = {
+        rowId(blank: false)
+        palletName(nullable: true)
+        boxName(nullable: true, validator: { String boxName, ImportPackingListItem item ->
+            if (boxName && !item.palletName) {
+                return ['packLevel1.required']
+            }
+            return true
+        })
+        lotNumber(nullable: true, blank: true, validator: { String lotNumber, ImportPackingListItem item ->
+            InventoryItem inventoryItem = item.product?.getInventoryItem(lotNumber)
+            if (inventoryItem) {
+                return true
+            }
+            return ['inventoryItemNotFound', lotNumber, item.product?.productCode]
+        })
+        expirationDate(nullable: true)
+        binLocation(nullable: true)
+        quantityPicked(min: 0, validator: { Integer quantityPicked, ImportPackingListItem item ->
+            if (!item.binLocationFound) {
+                return ['binLocationNotFound']
+            }
+            ProductAvailabilityService productAvailabilityService = Holders.grailsApplication.mainContext.getBean(ProductAvailabilityService)
+            InventoryItem inventoryItem = item.product?.getInventoryItem(item.lotNumber)
+            if (!inventoryItem) {
+                return ['inventoryItemNotFound']
+            }
+            Location currentLocation = AuthService.currentLocation
+            Integer quantity = productAvailabilityService.getQuantityAvailableToPromiseForProductInBin(inventoryItem, currentLocation, item.binLocation)
+            if (!quantity) {
+                return ['noStock']
+            }
+            if (quantityPicked > quantity) {
+                return ['overPick', quantityPicked]
```

Comment:
name this `overpick` since it's an actual word and not two words "over pick"). This is similar to "putaway". It's putaway, not "put away".

Reply:
Sources:
* https://www.merriam-webster.com/dictionary/overpick
* https://docs.oracle.com/en/cloud/saas/supply-chain-and-manufacturing/24b/famml/overpick-for-sales-orders-transfer-orders-and-work-orders.html#s20077043